src/lib/select/select.ts

```diff
@@ -656,12 +661,11 @@ export class MatSelect extends _MatSelectMixinBase implements AfterContentInit,
 
   /** Whether the select is in an error state. */
   get errorState(): boolean {
-    const isInvalid = this.ngControl && this.ngControl.invalid;
-    const isTouched = this.ngControl && this.ngControl.touched;
-    const isSubmitted = (this._parentFormGroup && this._parentFormGroup.submitted) ||
-        (this._parentForm && this._parentForm.submitted);
+    const ngControl = this.ngControl;
+    const parent = this._parentFormGroup || this._parentForm;
+    const matcher = this.errorStateMatcher || this._defaultErrorStateMatcher;
 
-    return !!(isInvalid && (isTouched || isSubmitted));
+    return ngControl && matcher.isErrorState(ngControl, parent);
```

Comment:
same, no need to check that `ngControl` is non-null

Reply:
Done.